pkg/build/api/types.go

```diff
@@ -25,6 +25,8 @@ const (
 	BuildJenkinsLogURLAnnotation = "openshift.io/jenkins-log-url"
 	// BuildJenkinsBuildURIAnnotation is an annotation holding a link to the Jenkins build
 	BuildJenkinsBuildURIAnnotation = "openshift.io/jenkins-build-uri"
+	// BuildMatchSourceURIAnnotation is an annotation on a Secret indicating the URIs for which it can be used
+	BuildMatchSourceURIAnnotation = "build.openshift.io/match-source-uri"
```

Comment:
I saw @deads2k's suggestion of this format, but given that all the rest of the build related annotations(seen just above this item) currently follow the form:  openshift.io/group.item, i'd like to suggest we do the same here, eg openshift.io/buildsecret.match-source-uri and at least retain some level of consistency with what we already have.

obviously this is ultimately up to @openshift/api-review 

either way i'd suggest the group name be "buildsecret" or "sourcesecret" instead of "build"